### ISSUE TITLE
Namecheap support

### ISSUE THREAD

**Issue by bviktor** [AnalogJ/lexicon#145]
Guys, what's the status of Namecheap support? I see there's a pretty old branch and even a prototype of sorts committed, is there any roadmap for this to happen? I'd really appreciate it :-)

**Comment by bviktor** [comment]
There's another fella hacking on the Namecheap API for Let's Encrypt support, it might be of help:

https://github.com/wdouglascampbell/dehydrated_namecheap_dns_api_hook/blob/master/namecheap_dns_api_hook.sh

Though I assume it's lack of time instead of lack of info :)

**Comment by AnalogJ** [comment]
Hey @bviktor ,

I think the issue with the Namecheap provider is that it requires a paid account, and I don't have one to test with. 

If you're willing to be a guinea pig, I'd love to merge in the namecheap branch.

**Comment by bviktor** [comment]
I'd gladly test it, how do I?

```
pip install dns-lexicon
git clone https://github.com/AnalogJ/lexicon.git
cd lexicon
git checkout namecheap
```

?

**Comment by AnalogJ** [comment]
Hey @bviktor , I had to fix an issue with the way that environmental variables were handled. 
It should be good to go now. If you want to fork off @pschmitt 's branch, and then add the integration test recordings, then I can definitely merge the PR.


The testing instructions are in the CONTRIBUTING.md file.

-Jason

**Comment by rbelnap** [comment]
I completed the test recordings, which required some modification to the provider itself to pass tests.  I forked off of the rebased version on https://github.com/polyzen/lexicon, and put in a PR there... but I'm afraid I am not familiar enough with git to see how to update existing PRs here... But the changes are available at https://github.com/rbelnap/lexicon/tree/Namecheap-rebase 

Hopefully there is enough info to get the code in the right place, my apologies if I've muddied the waters.  Let me know if there is anything else I can provide!

**Comment by AnalogJ** [comment]
`namecheap` support (with a small fix) was added in v2.1.18.

Thanks for pushing this through.

**Comment by bviktor** [comment]
Sorry for the late response. I've tried it today, it worked... for around the 5th time.

Before that I got a lot of "Challenge is invalid!" messages after a few successes. Tried to obtain a cert with 3 hostnames.

Could it be that Lexicon doesn't wait enough until the changes are propagated? I remember the other hook sometimes waiting for like 5-10 minutes, because Namecheap updates extremely slowly at times.
